phac-nml/irida-next
# Workflow Submission: Metadata field selection

## What does this PR do and why?
_Describe in detail what your merge request does and why._

This PR adds the ability for a user to select which metadata field to use for a metadata submission.

- Dropdown added to the sample sheet header for a given field
- Sample cell update with new metadata value once the field is changed.

## Screenshots or screen recordings
_Screenshots are required for UI changes, and strongly recommended for all other pull requests._

**Sample Sheet**
![image](https://github.com/phac-nml/irida-next/assets/11295750/10f7a244-4307-4697-94ff-8d086d23d65e)

**Request payload**
![image](https://github.com/phac-nml/irida-next/assets/11295750/e2fed2b8-ea05-4071-9711-2f90bfdb0bae)

## How to set up and validate locally
_Numbered steps to set up and validate the change are strongly suggested._

Add the following to `private/pipelines/phac-nml/iridanextexample/1.0.2/schema_input.json` after `reference_assembly`:
```json
            "insdc_accession": {
                "type": "string",
                "pattern": "^(SRR|ERR|DRR)\\S+$",
                "meta": ["insdc_accession"],
                "errorMessage": "Must provide a valid accession"
            },
            "country": {
                "type": "string",
                "meta": "country"
            }
````

- Login as user5
- Go to a [project with samples](http://localhost:3000/yersinia/yersinia-pseudotuberculosis/outbreak-2023/-/samples) (http://localhost:3000/yersinia/yersinia-pseudotuberculosis/outbreak-2023/-/samples)
- Select a couple of samples
- Click the launch pipeline button and select the first pipeline
- There should be two metadata columns `insdc_accession` and `country` that are select inputs which contain options that are all the metadata fields available in the current project.
- Changing the value of the select should automatically update the text in the column.

## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.
